adapters/Vercel.js

```
@@ -0,0 +1,102 @@
+import fetch from 'node-fetch';
+
+const vercelKey = (key, clientKey) => clientKey ? `${clientKey}-${key}` : key;
+
+export default class VercelAdapter {
+  constructor(logger, options) {
+    this.logger = logger;
+    this.options = options;
+  }
+
+  async get(key, clientKey) {
+    try {
+      const response = await fetch(
+        `${this.options.EDGE_CONFIG_API_URL}/${this.options.EDGE_CONFIG_ID}/item/${vercelKey(key, clientKey)}?token=${this.options.EDGE_CONFIG_TOKEN}`,
+      );
+      return await response.json();
+    } catch (error) {
```

What is the goal of catching the error at this level? It would be better to let the client function/method handle the exception or return a value that represents the error which could be handled by who is calling. This way you avoid "silent errors".

The same applies to all try/catch blocks in this module (file).

---

I got it. I allow to handle it on client side.